scripts/ktlint_lint_check.sh

```diff
@@ -14,7 +14,15 @@ else
     jar_file_path="$github_actions_path/oppia-android-tools/ktlint"
 fi
 
-java -jar $jar_file_path --android app/src/**/*.kt data/src/**/*.kt domain/src/**/*.kt testing/src/**/*.kt utility/src/**/*.kt
+committed_files=`git diff --name-only HEAD^ HEAD -- '***.kt'`
```

Comment:
A few questions:
1. Should we only be performing this check when running pre-push hooks? I think people may also use this to test their files prior to committing them.
2. Does this work both for files from several commits ago & files staged for commit?

Reply:
1. Yes. It is ideal to perform these checks only while running pre-push hooks. With the current change, users might not be able to check their `.kt` files before committing them. However, we can add an option **-c** to `ktlint_lint_check.sh`, and we can use this option in `pre-push.sh` while calling `ktlint_lint_check.sh` script to check only for committed files. 
When no arguments are specified, then we can run the ktlinter for all modules in our project.
What do you think ?

2. The command checks for files with changes between the current commit(HEAD) and the immediate previous commit(HEAD^). So, this does not work for files from several commits ago. 
And, this does not list the files staged for commit. It only lists out the files committed already.

Reply:
For point 1, we need to think of
1. Complexity for the contributor to understand these steps.
2. How important is this and what percentage of people will run the script just to check the file formatting?

Reply:
Thoughts:
1. Can we instead make the argument just for the pre-push hook? The default behavior should just lint the whole codebsae.
2. We should be designing this to compare against all files committed relative to develop so that it can work in a pre-push context

Reply:
@anandwana001 
1. For users, I think it will not make a difference. Only in the pre-push script, we can pass an option '-c' to make sure to check committed files only (which the users need not worry about). For regular users, when they run `ktlint_lint_check.sh` (without any additional arguments), it will proceed to check all the files in project (which is same as the existing behavior)
2. This could help us while working on projects that involve changes in a lot of files (which are not staged for commit). However, I think contributors might not always run the `ktlint_lint_check.sh` script separately, as they are notified of any lint errors by the `pre-push` script.

Reply:
@BenHenning 
1. Yes, this is what I meant to convey here https://github.com/oppia/oppia-android/pull/2725#discussion_r577342454. The default behavior will not be affected in this case.
2. I am not sure if this is achievable using git filters and commands. I will investigate more on this to check if it is possible to make this work in a pre-push context.